### PR TITLE
Remove duplicate lints, fix remaining ignored lint

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
Feel free to squash/merge and delete the branch if you're happy

**Comment by kevmoo** [comment]
@tvolkert  – the redness is just a slight dip in code coverage

**Comment by tvolkert** [comment]
We still want many of these lints.

**Comment by kevmoo** [comment]
> We still want many of these lints.

...and you get them! I'm removing the ones that were duplicated by `pkg:pedantic`

**Comment by tvolkert** [comment]
Then why the removal of the declared types?

**Comment by kevmoo** [comment]
> Then why the removal of the declared types?

Oh...because they are gratuitous. 😄  I guess this is the dart vs flutter style debate...

**Comment by kevmoo** [comment]
> > Then why the removal of the declared types?
> 
> Oh...because they are gratuitous. 😄 I guess this is the dart vs flutter style debate...

It makes migrating to null safety a lot easier if you don't have to touch types in so many places

File: bin/mesh_generator.dart [hunk 1]
```diff
@@ -82,13 +82,13 @@ void main(List<String> args) {
     print('');
     return;
   }
-  final String type = fixedArgs.removeAt(0);
-  final GenerateFunction generator = generators[type];
+  final type = fixedArgs.removeAt(0);
+  final generator = generators[type];
```
Comment (tvolkert) [inline bin/mesh_generator.dart]: Case in point: I have no idea what type these are now looking at the code.

File: lib/src/vector_math/aabb2.dart [hunk 2]
```diff
@@ -136,8 +136,8 @@ class Aabb2 {
 
   /// Return if this contains [other].
   bool containsAabb2(Aabb2 other) {
-    final Vector2 otherMax = other._max;
-    final Vector2 otherMin = other._min;
+    final otherMax = other._max;
```
Comment (tvolkert) [inline lib/src/vector_math/aabb2.dart]: or here

File: lib/src/vector_math/aabb3.dart [hunk 3]
```diff
@@ -159,19 +159,19 @@ class Aabb3 {
     ray..copyAt(_min, limitMin)..copyAt(_max, limitMax);
 
     if (_max.x < _min.x) {
-      final double temp = _max.x;
+      final temp = _max.x;
```
Comment (tvolkert) [inline lib/src/vector_math/aabb3.dart]: or here.

**Comment by tvolkert** [comment]
I found a few examples quickly where the removal of the type makes the code less readable, and this code is read _way_ more often than it's written.

File: bin/mesh_generator.dart (hunk 1 above)
Comment (kevmoo) [reply bin/mesh_generator.dart]: click to navigate to `generators` definition?

*sigh* Yeah, this is a subjective things. Seattle folks generally don't like the noise. 🤷‍♂

**Comment by tvolkert** [comment]
Clicking to navigate implies you're in an IDE, which is often not the case - making type declarations very much not noise.

@johnmccutchan if you're ok with the change, feel free to LGTM.  I'm only a faux owner of the package 🙂

**Comment by johnmccutchan** [comment]
I am a big fan of explicit types.

**Comment by kevmoo** [comment]
I'll roll back

**Pull request closed without merging**